crates/bindings/src/lib.rs

```diff
@@ -1030,6 +1032,50 @@ impl ReducerContext {
     pub fn as_read_only(&self) -> ViewContext {
         ViewContext::new(self.sender)
     }
+
+    ///  Create a new UUIDv4 using the built-in RNG.
+    //// # Example
+    /// ```no_run
+    /// # #[cfg(target_arch = "wasm32")] mod demo {
+    /// use spacetimedb::{reducer, ReducerContext, Uuid};
+    ///
+    /// #[reducer]
+    /// fn generate_uuid_v4(ctx: &ReducerContext) -> Uuid {
+    ///     ctx.new_uuid_v4()
+    /// }
+    /// # }
+    /// ```
+    pub fn new_uuid_v4(&self) -> anyhow::Result<Uuid> {
+        let mut bytes = [0u8; 16];
+        self.rng().try_fill_bytes(&mut bytes)?;
+        Ok(Uuid::from_random_bytes_v4(bytes))
+    }
+
+    /// Create a new UUIDv7 using the provided [`ClockGenerator`].
+    ///
+    /// **NOTE**: To maintain UUIDv7 monotonicity guarantees, do not use this method
+    /// from multiple threads or contexts sharing the same [`ClockGenerator`].
+    ///
+    /// Prefer to create a single [`ClockGenerator`] per context and reuse it for all UUIDv7 generation for the same `table` or operation.
+    ///
+    /// # Example
+    /// ```no_run
+    /// # #[cfg(target_arch = "wasm32")] mod demo {
+    /// use spacetimedb::{reducer, ReducerContext, ClockGenerator, Uuid};
+    ///
+    /// #[reducer]
+    /// fn generate_uuid_v7(ctx: &ReducerContext) -> Result<Uuid, Box<dyn std::error::Error>> {
+    ///     let mut clock = ClockGenerator::new(ctx.timestamp);
+    ///     let uuid = ctx.new_uuid_v7(&mut clock)?;
+    ///     Ok(uuid)
+    /// }
+    /// # }
+    /// ```
+    pub fn new_uuid_v7(&self, clock: &mut ClockGenerator) -> anyhow::Result<Uuid> {
```

Comment:
Why don't we just stash a `ClockGenerator` (in a `RefCell` or similar) inside the `ReducerContext`? Why is this part of the public API?

Reply:
Also, are we concerned that reducers may run faster than 1/ms, resulting in a later TX seeing as its base the same timestamp as a previous TX generated as its second step?